packages/legends/src/types.ts

```diff
@@ -0,0 +1,143 @@
+import React from 'react'
+import { SymbolProps } from './svg/symbols/types'
+import { CompleteTheme } from '@nivo/core'
+
+/**
+ * This can be used to add effect on legends on interaction.
+ */
+type EffectProps = {
+    on: 'hover'
+    style: Partial<{
+        itemTextColor: string
+        itemBackground: string
+        itemOpacity: number
+        symbolSize: number
+        symbolBorderWidth: number
+        symbolBorderColor: string
+    }>
+}
+
+type SymbolShape = 'circle' | 'diamond' | 'square' | 'triangle'
+
+type BoxLegendSymbolProps = Partial<{
+    symbolShape: SymbolShape | React.FC<SymbolProps>
+    symbolSize: number
+    symbolSpacing: number
+    symbolBorderWidth: number
+    symbolBorderColor: string
+}>
+
+type InteractivityProps = Partial<
+    Record<
+        'onClick' | 'onMouseEnter' | 'onMouseLeave',
+        (datum: Datum, event: React.MouseEvent<SVGRectElement>) => void
+    >
+>
+
+export type LegendAnchor =
+    | 'top'
+    | 'top-right'
+    | 'right'
+    | 'bottom-right'
+    | 'bottom'
+    | 'bottom-left'
+    | 'left'
+    | 'top-left'
+    | 'center'
+
+export type LegendDirection = 'column' | 'row'
+
+export type LegendItemDirection =
+    | 'left-to-right'
+    | 'right-to-left'
+    | 'top-to-bottom'
+    | 'bottom-to-top'
+
+export type Datum = {
+    id: string | number
+    label: string | number
+    color?: string
+    fill?: string
+}
+
+type CommonLegendProps = {
+    direction: LegendDirection
+    padding: number | Record<'top' | 'right' | 'bottom' | 'left', number>
```

Comment:
Similarly, I never used the padding prop before and can't find any for of documentation describing how/what it is used for. Should this really be required?